src/utils/bigNumbers.ts

```diff
@@ -23,3 +23,21 @@ export const betweenZeroAndOne = (amount: BigNumber) => {
 export const invertPermyriad = (permyriad: BigNumber) => {
   return BigNumber.from(10000).sub(permyriad)
 }
+
+/**
```

Comment:
This is a weird thing to start with, but a necessity for web3-onboard. I think this comes from metamask, but it is very fussy about things not having padded zeros. For instance, hex of 4 (via bignumber) gets padded as 0x04, which it complains about. This uttility script strips the extra 0s (if any)